File: IsraelHiking.Web/android/app/src/main/java/com/mapeak/car/MapeakCarAppService.java
package com.mapeak.car;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.car.app.CarAppService;
import androidx.car.app.Session;
import androidx.car.app.validation.HostValidator;

public class MapeakCarAppService extends CarAppService {
    public static final String LOG_TAG = "MapeakCarAppService";

    @NonNull
    @Override
    public Session onCreateSession() {
        Log.d(LOG_TAG, "Session created");
        return new CarSession();
    }

    @Override
    public void onDestroy() {
        Log.d(LOG_TAG, "onDestroy");
        super.onDestroy();
    }

    @NonNull
    @Override
    public HostValidator createHostValidator() {
        // HM TODO: bring this back
        // if (isDebug) {
        return HostValidator.ALLOW_ALL_HOSTS_VALIDATOR;
        // } else {
        // return new HostValidator.Builder(this)
        // .addAllowedHosts(R.array.car_template_hosts_allowlist)
        // .build();
        // }
    }
}

File: IsraelHiking.Web/android/app/src/main/java/com/mapeak/car/MapeakCarAppService.kt
package com.mapeak.car

import android.content.pm.ApplicationInfo
import android.util.Log
import androidx.car.app.CarAppService
import androidx.car.app.R
import androidx.car.app.Session
import androidx.car.app.validation.HostValidator

class MapeakCarAppService : CarAppService() {
    override fun onCreateSession(): Session {
        Log.d(LOG_TAG, "Session created")
        return CarSession()
    }

    override fun onDestroy() {
        Log.d(LOG_TAG, "onDestroy")
        super.onDestroy()
    }

    override fun createHostValidator(): HostValidator {
        val isDebuggable = (applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE) != 0
        return if (isDebuggable) {
            HostValidator.ALLOW_ALL_HOSTS_VALIDATOR
        } else {
            HostValidator.Builder(applicationContext)
                    .addAllowedHosts(R.array.hosts_allowlist_sample)
                    .build()
        }
    }

    companion object {
        const val LOG_TAG: String = "MapeakCarAppService"
    }
}
